client/app/scripts/superdesk/editor/editor.js

```diff
@@ -199,11 +205,171 @@ function FindReplaceCommand(rootNode) {
     }
 }
 
+function SpellcheckService($q, dictionaries) {
+
+    var selection = document.getSelection(),
+        dict, dictPromise;
+
+    function getDict() {
+        if (dict) {
+            return $q.when(dict);
+        }
+
+        if (!dictPromise) {
+            dictPromise = dictionaries.fetch().then(function(result) {
+                return dictionaries.open(result._items[0]);
+            }).then(function(_dict) {
+                dict = _.zipObject(_dict.content);
+                return dict;
+            });
+        }
+
+        return dictPromise;
+    }
+
+    this.check = function check(text) {
+        return getDict().then(function() {
+            var words = _.uniq(_.filter(text.split(/[\W0-9]+/)));
```

Comment:
Adding a marker comment (splitting non-ascii words) - the last one got outdated due to the obsolete diff.